packages/use-dataloader/src/DataLoaderProvider.tsx

```diff
@@ -99,70 +165,84 @@ const DataLoaderProvider = ({ children, cacheKeyPrefix, onError }: {
       if (key && typeof key === 'string') {
         setCachedData(actualCachedData => {
           const tmp = actualCachedData
-          delete tmp[key]
+          delete tmp[computeKey(key)]
 
           return tmp
         })
       }
     },
-    [setCachedData],
+    [setCachedData, computeKey],
   )
   const clearAllCachedData = useCallback(() => {
     setCachedData({})
   }, [setCachedData])
 
-  const reload = useCallback(async (key?: string) => {
-    if (key && typeof key === 'string') {
-      await (reloads.current[key] && reloads.current[key]())
-    }
-  }, [])
+  const reload = useCallback(
+    async (key?: string) => {
+      if (key && typeof key === 'string') {
+        await (reloads.current[computeKey(key)] &&
+          reloads.current[computeKey(key)]())
+      }
+    },
+    [computeKey],
+  )
 
   const reloadAll = useCallback(async () => {
     await Promise.all(
       Object.values(reloads.current).map(reloadFn => reloadFn()),
     )
   }, [])
 
-  const getCachedData = useCallback((key?: string) => {
-    if (key) {
-      return cachedData.current[key] || undefined
-    }
+  const getCachedData = useCallback(
+    (key?: string) => {
+      if (key) {
+        return cachedData[computeKey(key)] || undefined
+      }
 
-    return cachedData.current
-  }, [])
+      return cachedData
+    },
+    [computeKey, cachedData],
+  )
 
-  const getReloads = useCallback((key?: string) => {
-    if (key) {
-      return reloads.current[key] || undefined
-    }
+  const getReloads = useCallback(
+    (key?: string) => {
+      if (key) {
+        return reloads.current[computeKey(key)] || undefined
+      }
```

Comment:
To put a throw error on every function if key is not string or it doesn't exists